pkg/keystorev3/walletfile.go

```diff
@@ -76,11 +86,21 @@ type cryptoPbkdf2 struct {
 	KDFParams kdfParamsPbkdf2 `json:"kdfparams"`
 }
 
-type walletFileBase struct {
+type walletFileCoreFields struct {
+	ID      *fftypes.UUID `json:"id"`
+	Version int           `json:"version"`
+}
+
+type walletFileMetadata struct {
+	// address is not technically part of keystorev3 syntax, and note this can be overridden/removed by callers of the package
 	Address ethtypes.AddressPlainHex `json:"address"`
-	ID      *fftypes.UUID            `json:"id"`
-	Version int                      `json:"version"`
+	// arbitrary additional fields that can be stored in the JSON, including overriding/removing the "address" field (other core fields cannot be overridden)
+	metadata map[string]interface{}
+}
 
+type walletFileBase struct {
+	walletFileCoreFields
+	walletFileMetadata
 	privateKey []byte
 	keypair    *secp256k1.KeyPair
```

Comment:
would be great to have this removed as well, to make the walletFile object agnostic to the public key algorithms. this was done in #69. but no need to hold the merge for this.

Reply:
Yes - agreed... I wonder if I can make it calculate-on-demand as a way to keep the interface identical 🤔

Reply:
great suggestion @jimthematrix - I've managed to do that I think completely removing the field, and even actually removing the `Address` struct field.

I realized we don't expose the field directly anyway.

Mind taking another look?